src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java

```diff
@@ -184,6 +190,25 @@ private void initIdpCerts(IDPSSODescriptor idpSsoDescriptor, HashMap<String, Obj
         }
     }
 
+    private void initSpSignaturePrivateKey(Settings settings, HashMap<String, Object> configProperties, Path configPath) throws SamlConfigException {
+        try {
+            PrivateKey result = PemKeyReader.loadKeyFromStream(settings.get("sp.signature_private_key_password"),
```

Comment:
I don't see why is it necessary to load key every time here instead of doing it once as it is done.

Reply:
We need to add the signature private key to the config properties which become the Saml2Settings. This is so that it can be accessed by the onelogin SamlResponse here: https://github.com/onelogin/java-saml/blob/master/core/src/main/java/com/onelogin/saml2/authn/SamlResponse.java#L1074

Reply:
Why can't it be passed to the `Saml2SettingsProvider.java` constructor?

Reply:
Makes sense. And that makes this a much smaller PR.